app/main/views/root.py

```diff
@@ -68,13 +71,41 @@ def login():
     form_type = metadata["form_type"]
 
     logger.debug("Requested questionnaire %s for form type %s", eq_id, form_type)
+
     if not eq_id or not form_type:
         logger.error("Missing EQ id %s or form type %s in JWT", eq_id, form_type)
         raise NotFound
 
-    json, schema = get_schema()
-
-    navigator = get_navigator(json)
-    current_location = navigator.get_latest_location(get_answers(current_user), get_completed_blocks(current_user))
-
-    return redirect('/questionnaire/' + eq_id + '/' + form_type + '/' + collection_id + '/' + current_location)
+    json, _ = get_schema()
+
+    navigator = Navigator(json, get_answer_store(current_user))
+    current_location = navigator.get_latest_location(get_completed_blocks(current_user))
+
+    if current_location['block_id'] == 'introduction':
```

Comment:
It might be tidy to condense to the below?

```
block = current_location['block_id']
if block in ['introduction'...]:
    return redirect(url_for('questionnaire.get_' + block,
                                 eq_id=eq_id,
                                 form_type=form_type,
                                 collection_id=collection_id))
```